toolchain/lower/testdata/impl/extend_impl.carbon

```diff
@@ -31,20 +31,20 @@ fn InstanceAccess(a: A) -> i32 {
 // CHECK:STDOUT: ; ModuleID = 'extend_impl.carbon'
 // CHECK:STDOUT: source_filename = "extend_impl.carbon"
 // CHECK:STDOUT:
-// CHECK:STDOUT: define i32 @F() !dbg !4 {
+// CHECK:STDOUT: define i32 @"_CF.A.Main:I.Main.A.Main"() !dbg !4 {
```

Comment:
This mangling looks a little strange -- specifically repeating the `.A.Main` part twice. It also looks like the trailing portion might be ambiguous. How should we read this?

Reply:
Ah, yep - that's a bug.

Fixed.

The issue was that it was mangling in the parent name scopes, including the parent name scope of wherever the impl was defined - which, if I understand correctly, isn't relevant. (an impl is uniquely defined by its interface and constraint, regardless of the scope in which that impl is defined - ie: two impls of the same interface+constraint, written in different namespaces, are invalid/do collide. (well, I guess there's no syntax to define an impl in a namespace, so it only happens to collide between, I guess, two files both containing an impl or a single file containing an inline impl inside a class definition, and one outside, which fails with:
```
basic.carbon:11:1: ERROR: Redefinition of `impl B as A`.
impl B as A {
^~~~~~~~~~~~~
basic.carbon:6:3: Previous definition was here.
  extend impl as A {
  ^~~~~~~~~~~~~~~~~~
```

Fixed this by adding the lexical parent scope after the switch, and having the impl handling `continue` resulting in the parent scope addition being skipped.

Now there shouldn't be any trailing portion after an impl, it should be `<function name>.<self name>:<constraint name>`.